dipy/workflows/reconst.py

```diff
@@ -21,6 +21,330 @@
 from dipy.reconst.shm import CsaOdfModel
 from dipy.workflows.workflow import Workflow
 
+from dipy.reconst import mapmri
+
+class ReconstMAPMRIFlow(Workflow):
+    @classmethod
+    def get_short_name(cls):
+        return 'mapmri'
+
+    def run(self, data_file, data_bvecs, data_bvals, model_type='', out_rtop='rtop.nii.gz',
+            out_lapnorm='lapnorm.nii.gz',out_msd='msd.nii.gz', out_qiv='qiv.nii.gz', out_rtap='rtap.nii.gz',
+            out_rtpp='rtpp.nii.gz', small_delta=0.0129, big_delta=0.0218, save_metrics=[], out_dir=''):
+        """ Workflow for the app-dipy-mapmri on Brain-Life (www.brain-life.org). 
+        Generates rtop, lapnorm, msd, qiv, rtap, rtpp saved in a nifti format in input files provided by
+        `data_file` and saves the png file to an output directory specified by 
+        `out_dir`.
+
+        Parameters
+        ----------
+        data_file : string
+            Path to the input volume.
+        data_bvecs : string
+            Path to the bvec files.
+        data_bvals : 
+            Path to the bval files.
+        small_delta :
+            Small delta value used in generation of gradient table of provided
+            bval and bvec. (default: 0.0129)
+        big_delta :
+            Big delta value used in generation of gradient table of provided 
+            bval and bvec. (default: 0.0218)
+        model_type : string
+            Model type to fit.
+            Possible values: laplacian, positivity, both
+            (default: both)
+        save_metrics :
+            List of metrics to save.
+            Possible values: rtop, laplacian_signal, msd, qiv, rtap, rtpp
+            (default: [] (all))
+        out_dir : string, optional
+            Output directory (default: input file directory)
+        out_rtop : string, optional
+            Name of the rtop to be saved
+        out_lapnorm : string, optional
+            Name of the norm of laplacian signal to be saved
+        out_msd : string, optional
+            Name of the msd to be saved
+        out_qiv : string, optional
+            Name of the qiv to be saved
+        out_rtap : string, optional
+            Name of the rtap to be saved
+        out_rtpp : string, optional
+            Name of the rtpp to be saved
+        """
+        io_it = self.get_io_iterator()
+        for dwi, bval, bvec, out_rtop, out_lapnorm, out_msd, out_qiv, out_rtap, out_rtpp in io_it:
+
+            logging.info('Computing DTI metrics for {0}'.format(dwi))
+            img = nib.load(dwi)
+            data = img.get_data()
+            affine = img.affine
+            bvals, bvecs = read_bvals_bvecs(bval, bvec)
+
+            gtab = gradient_table(bvals=bvals, bvecs=bvecs,
+                                  small_delta=small_delta,
+                                  big_delta=big_delta, b0_threshold=50)
```

Comment:
Why is the b0_threshold here set to 50?

Reply:
I believe that `b0_threshold` should be an input, like here: https://github.com/nipy/dipy/blob/master/dipy/workflows/reconst.py#L30